CollapseLauncher/Classes/Helper/Database/DBHandler.cs

```diff
@@ -0,0 +1,304 @@
+            catch (Exception e) when (!redirectThrow)
+            {
+                LogWriteLine($"[DBHandler::Init] Error when (re)initializing database system!\r\n{e}", LogType.Error, true);
+            }
+            catch (Exception e)
+            {
+                LogWriteLine($"[DBHandler::Init] Error when (re)initializing database system!\r\n{e}", LogType.Error, true);
+                throw;
+            }
+        }
+
+        private static void Dispose()
+        {
+            _database = null;
+            _token = null;
+            _uri = null;
+            _userId = null;
+            _userIdHash = null;
+        }
+
+        public static async Task<string> QueryKey(string key, bool redirectThrow = false)
+        {
+            if (!IsEnabled) return null;
+        #if DEBUG
+            var r   = new Random();
+            var sId = Math.Abs(r.Next(0, 1000).ToString().GetHashCode());
+            LogWriteLine($"[DBHandler::QueryKey][{sId}] Invoked!\r\n\tKey: {key}", LogType.Debug, true);
+            var t = System.Diagnostics.Stopwatch.StartNew();
+        #endif
+            const int retryCount = 3;
+            for (var i = 0; i < retryCount; i++)
+            {
+                try
+                {
+                    // Get table row for exact key
+                    var rs =
+                        await
+                            _database
+                               .Execute($"SELECT value FROM \"uid-{_userIdHash}\" WHERE key = ?", key);
+                    if (rs != null)
+                    {
+                        // freaking black magic to convert the column row to the value 
+                        var str =
+                            string.Join("", rs.Rows.Select(row => string.Join("", row.Select(x => x.ToString()))));
+                    #if DEBUG
+                        LogWriteLine($"[DBHandler::QueryKey][{sId}] Got value!\r\n\tKey: {key}\r\n\tValue:\r\n{str}", LogType.Debug,
+                                     true);
+                    #endif
+                        return str;
+                    }
+                }
+                catch (LibsqlException ex) when ((ex.Message.Contains("STREAM_EXPIRED") ||
+                                                  ex.Message.Contains("Received an invalid baton")) &&
+                                                 i < retryCount - 1)
+                {
+                    if (i > 0)
+                        LogWriteLine("[DBHandler::QueryKey] Database stream expired, retrying...", LogType.Error, true);
+
+                    await Init();
+                }
+                catch (Exception ex) when (i < retryCount - 1)
+                {
+                    LogWriteLine($"[DBHandler::QueryKey] Failed when getting value for key {key}! Retrying...\r\n{ex}",
+                                 LogType.Error, true);
+                    break;
+                }
+                catch (Exception ex) when (!redirectThrow)
```

Comment:
Copied my response from other thread 
> When `redirectThrow` is set to true, any exception will be thrown back to the method caller. This is (currently) only used in the settings validation as I need a way to get what errors to be displayed to the user. The only difference is just it has `throw` instead of `return`